src/rgw/rgw_asio_frontend.cc

```diff
@@ -292,7 +293,7 @@ void handle_connection(boost::asio::io_context& context,
       }
     }
 
-    if (!parser.keep_alive()) {
+    if (!parser.keep_alive() || http_ret != 200) {
```

Comment:
i don't think we want to close client connections on any non-200 response. the `while (!parser.is_done())` loop just below is meant to recover from this - can you tell why that fails here?

Reply:
ah, the tracker issue said this was due to interaction with "Expect: 100-continue". i'm testing a fix for that in https://github.com/ceph/ceph/pull/49642